### PR TITLE
GEOPY-1528: Add UIJson group to API

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-1528 - Add UIJson group to API**

### CODE REVIEW

File: geoh5py/groups/uijson_group.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        if getattr(self, "_options", None) is None:
+            self._options = self.workspace.fetch_metadata(self.uid, argument="options")
+
+        if self._options is None:
+            self._options = {}
+
+        return self._options
+
+    @options.setter
+    def options(self, value: dict | None):
```
Comment (benk-mira) [inline geoh5py/groups/uijson_group.py:58]: Is there any reason to accept None here?

**Review by benk-mira — CHANGES_REQUESTED** [review]
Nice!  couple suggestions

File: geoh5py/groups/uijson_group.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#  Copyright (c) 2024 Mira Geoscience Ltd.
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/groups/uijson_group.py:1]: I agree. Can probably do the change across.

File: geoh5py/groups/uijson_group.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    """Group for SimPEG inversions."""
+
+    __TYPE_UID = uuid.UUID("{BB50AC61-A657-4926-9C82-067658E246A0}")
+
+    _name = "UIJson"
+    _description = "UIJson"
+    _options = None
+
+    def __init__(self, group_type: GroupType, name="UIJson", **kwargs):
+        assert group_type is not None
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/groups/uijson_group.py:34]: True. Changing to a raise on base class and removing everywhere else.

File: geoh5py/groups/uijson_group.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        """
+        Metadata attached to the entity.
+        """
+        if getattr(self, "_options", None) is None:
+            self._options = self.workspace.fetch_metadata(self.uid, argument="options")
+
+        if self._options is None:
+            self._options = {}
+
+        return self._options
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/groups/uijson_group.py:55]: Had to change some mechanics in the writer, but makes sense.

File: geoh5py/groups/uijson_group.py (hunk 1 above)
Comment (domfournier) [reply geoh5py/groups/uijson_group.py:58]: It gets rid of options on file if set to None.

**Review by benk-mira — APPROVED** [review]
Looks good, thanks for the updates

**Pull request merged**